src/main/java/dev/openfeature/sdk/multiprovider/MultiProvider.java

```diff
@@ -0,0 +1,150 @@
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/** <b>Experimental:</b> Provider implementation for Multi-provider. */
+@Slf4j
+public class MultiProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "multiprovider";
+
+    public static final int INIT_THREADS_COUNT = 8;
+    private final Map<String, FeatureProvider> providers;
+    private final Strategy strategy;
+    private MultiProviderMetadata metadata;
+
+    /**
+     * Constructs a MultiProvider with the given list of FeatureProviders, by default uses FirstMatchStrategy.
+     *
+     * @param providers the list of FeatureProviders to initialize the MultiProvider with
+     */
+    public MultiProvider(List<FeatureProvider> providers) {
+        this(providers, null);
+    }
+
+    /**
+     * Constructs a MultiProvider with the given list of FeatureProviders and a strategy.
+     *
+     * @param providers the list of FeatureProviders to initialize the MultiProvider with
+     * @param strategy the strategy
+     */
+    public MultiProvider(List<FeatureProvider> providers, Strategy strategy) {
+        this.providers = buildProviders(providers);
+        if (strategy != null) {
+            this.strategy = strategy;
+        } else {
+            this.strategy = new FirstMatchStrategy();
+        }
+    }
+
+    protected static Map<String, FeatureProvider> buildProviders(List<FeatureProvider> providers) {
+        Map<String, FeatureProvider> providersMap = new LinkedHashMap<>(providers.size());
+        for (FeatureProvider provider : providers) {
+            FeatureProvider prevProvider =
+                    providersMap.put(provider.getMetadata().getName(), provider);
+            if (prevProvider != null) {
+                log.warn("duplicated provider name: {}", provider.getMetadata().getName());
+            }
+        }
+        return Collections.unmodifiableMap(providersMap);
+    }
+
+    /**
+     * Initialize the provider.
+     *
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        var metadataBuilder = MultiProviderMetadata.builder();
+        metadataBuilder.name(NAME);
+        HashMap<String, Metadata> providersMetadata = new HashMap<>();
+        ExecutorService initPool = Executors.newFixedThreadPool(Math.min(INIT_THREADS_COUNT, providers.size()));
```

Comment:
```suggestion
        ExecutorService executorService = Executors.newFixedThreadPool(Math.min(INIT_THREADS_COUNT, providers.size()));
```
`initPool` seems odd, we should properly name it, and i suggest to name it `executorService`